### ISSUE TITLE
[Bug]: wandb yolov8 integration showing unused COCO class labels

### ISSUE THREAD

**Issue by jayktee** [wandb/wandb#8598]
## Bug
I am doing a transfer-learning training using the pre-trained COCO YOLOv8 checkpoint. I create a custom `data.yaml` which passes 7 labels to predict. A `print(model.names)` also shows the expected (not COCO) classes.

The logs correctly show:
   `4 Overriding model.yaml nc=80 with nc=7`

However, when I check the produced `Validation-Table` in the WandB web app, I see that the Validation-Table table images are showing other COCO class labels that were excluded during training, with their original COCO names. 

My 7 classes are named and labeled with bounding-boxes accurately in the `Validation-Table` but there are also bounding-boxes with unused labels. See screenshot below.

This is a known issue over in the `ultralytics` Github issues page (see [here](https://github.com/ultralytics/ultralytics/issues/13987) and [here](https://github.com/ultralytics/ultralytics/issues/13534)).

It could be an issue in the `add_wandb_callback()` function?

## Code

```
wandb.init(project=config["project"], entity=config["entity"])

# Load a model
model = YOLO(config["model"])

# Set up callbacks
add_wandb_callback(model, enable_model_checkpointing=True)

# Train the model
train_results = model.train(
        data=config["data"],
        epochs=config["epochs"],
        imgsz=config["imgsz"],
        device=config["device"],
        project=config["model_save_dir"],
        # callbacks=[custom_callback],
    )

model.val()

model(image_file_paths)

```

## Environment
WandB version:
0.16.6
OS:
Apple M1
Python version:
3.10.14
Versions of relevant libraries:
ultralytics 8.3.9
YOLO v8

![Image](https://github.com/user-attachments/assets/90815125-fec5-483c-a9af-1558123f33c7)

**Comment by jayktee** [comment]
hi! following up on this issue. Has anyone on the wandb side been able to reproduce? Thanks!

**Comment by jayktee** [comment]
Hi Jason, 

Thanks for checking in. We did try the code using the following library versions but the problem still persists. Thanks!

ultralytics==8.0.238 and wandb==0.18.3
